parsec/backend/drivers/postgresql/user.py

```diff
@@ -94,140 +102,164 @@ def __init__(self, dbh, *args):
-        )
+                await conn.execute(
+                    """INSERT INTO users (user_id, created_on, created_by, broadcast_key)
+                    VALUES ($1, $2, $3, $4)
+                    """,
+                    user_id,
+                    now,
+                    author,
+                    broadcast_key,
+                )
+                await conn.executemany(
+                    """INSERT INTO user_devices (user_id, device_name, created_on, verify_key, revocated_on)
+                    VALUES ($1, $2, $3, $4, NULL)
+                    """,
+                    [(user_id, name, now, key) for name, key in devices],
+                )
 
     async def get(self, user_id):
-        try:
-            created_on, created_by, broadcast_key = await self.dbh.fetch_one(
-                "SELECT created_on, created_by, broadcast_key FROM users WHERE user_id = %s",
-                (user_id,),
-            )
-        except (TypeError, ValueError):
-            raise NotFoundError(user_id)
-
-        user = {
-            "user_id": user_id,
-            "broadcast_key": broadcast_key.tobytes(),
-            "created_by": created_by,
-            "created_on": pendulum.from_timestamp(created_on),
-            "devices": {
-                d_name: {
-                    "created_on": pendulum.from_timestamp(d_created_on),
-                    "configure_token": d_configure_token,
-                    "verify_key": d_verify_key.tobytes() if d_verify_key else None,
-                    "revocated_on": (
-                        pendulum.from_timestamp(d_revocated_on) if d_revocated_on else None
-                    ),
+        async with self.dbh.pool.acquire() as conn:
+            async with conn.transaction():
+                try:
+                    created_on, created_by, broadcast_key = await conn.fetchrow(
+                        "SELECT created_on, created_by, broadcast_key FROM users WHERE user_id = $1",
+                        user_id,
+                    )
+                except (TypeError, ValueError):
+                    raise NotFoundError(user_id)
+
+                user = {
+                    "user_id": user_id,
+                    "broadcast_key": broadcast_key,
+                    "created_by": created_by,
+                    "created_on": pendulum.from_timestamp(created_on),
+                    "devices": {
+                        d_name: {
+                            "created_on": pendulum.from_timestamp(d_created_on),
+                            "configure_token": d_configure_token,
+                            "verify_key": d_verify_key if d_verify_key else None,
+                            "revocated_on": (
+                                pendulum.from_timestamp(d_revocated_on) if d_revocated_on else None
+                            ),
+                        }
+                        for d_name, d_created_on, d_configure_token, d_verify_key, d_revocated_on in await conn.fetch(
```

Comment:
Ça devient gros comme comprehension dict, je pense qu'il faudrait mieux faire ça avec une boucle for classique à part